### PR TITLE
Setup devise

### PR DESCRIPTION
**Pull request by lorenzoworx**
## In this pull request @ProgramKingAbel and I have implemented the following changes:
- Remove `current_user` method in `ApplicationController`
- Install and setup devise
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- Modify the views in devise for registration and login to match the wireframes from the Sneak Peek

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -45,12 +45,28 @@
   end
```
Comment (Munsa1) [inline db/schema.rb:46]: - [x] Due to insufficient files changed I will leave a comment here. Kindly delete the seed data in your `seed.rb` file as it is  causing an error when migrating some files.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Munsa1) [inline app/models/user.rb]: - [x] The application should ask for confirmation of email,  `:confirmable` should enabled in the user model, kindly do so.
- [x] Consider removing commented lines 2 and 3 for a cleaner looking codebase

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
```
Comment (Munsa1) [inline app/views/devise/sessions/new.html.erb]: - [x] It is a requirement that your design for the login and sign up page follows the given wireframe, kindly do so

Given frame
image

Your design
image

- Kindly fix the Login button to span the same width as the input fields so as to match the given design

#### Note that this applies to the sign up page as well the `Password` confirmation and the sign up button should span the same width as the other input fields above them

image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @lorenzoworx @ProgramKingAbel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Devise Setup Correctly :heavy_check_mark: 
Created Views For Devise :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 
Descriptive Readme File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb (hunk 1 above)
Comment (lorenzoworx) [reply db/schema.rb:46]: 👍🏽👌🏾

Comment (Whoistolu) [reply db/schema.rb:46]: - well resolved 👍🏽

File: app/models/user.rb (hunk 2 above)
Comment (Whoistolu) [reply app/models/user.rb]: - Well resolved 👍🏽

File: app/views/devise/sessions/new.html.erb (hunk 3 above)
Comment (Whoistolu) [reply app/views/devise/sessions/new.html.erb]: - Well resolved 
Screenshot 2023-09-15 at 18 42 05

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**